django/contrib/admin/utils.py

```diff
@@ -506,11 +516,19 @@ def construct_change_message(form, formsets, add):
                         }
                     })
                 for changed_object, changed_fields in formset.changed_objects:
+                    changed_fields_labels = []
+                    for field_name in changed_fields:
+                        try:
+                            verbose_field_name = str(formset.forms[0].fields[field_name].label) or field_name
+                        except KeyError:
+                            verbose_field_name = field_name
```

Comment:
This `except` block is never tested in the `formsets` case.

Reply:
Fixed :)